File: Tree/102.Binary-Tree-Level-Order-Traversal/102.Binary-Tree-Level-Order-Traversal.ts
function levelOrder(root: TreeNode | null): number[][] {
  if (root === null) {
    return [];
  }
  const queue = [root];
  const ret: number[][] = [];
  while (queue.length !== 0) {
    const levelLen = queue.length;
    const level = [];
    for (let i = 0; i < levelLen; i++) {
      const node = queue.shift();
      level.push(node.val);
      if (node.left) {
        queue.push(node.left);
      }

      if (node.right) {
        queue.push(node.right);
      }
    }
    ret.push(level);
  }
  return ret;
}

File: Tree/102.Binary-Tree-Level-Order-Traversal/102.Binary-Tree-Level-Order-Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        result = []
        if not root:
            return result

        queue = [root]
        while len(queue) > 0:
            currentLevel = []
            currentLevelLength = len(queue)
            for i in range(currentLevelLength):
                node = queue.pop(0)
                currentLevel.append(node.val)

                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            result.append(currentLevel)
        return result
